tests/test_is_locking_cvx.py

```diff
@@ -0,0 +1,74 @@
+import brownie
+from brownie import MockToken, accounts, chain, Wei, interface, history
+from helpers.constants import MaxUint256
+from helpers.SnapshotManager import SnapshotManager
+from helpers.time import days
+from config.badger_config import sett_config
+import pytest
+from conftest import deploy
+from rich.console import Console
+
+console = Console()
+
+@pytest.mark.parametrize(
+    "sett_id",
+    sett_config.helpers,
+)
+def test_deposit_withdraw_single_user_flow(sett_id):
+    if sett_id == "cvx":
+        return 
+    # Setup
+    deployed = deploy(sett_config.helpers[sett_id])
+
+    deployer = deployed.deployer
+    sett = deployed.sett
+    want = deployed.want
+    strategy = deployed.strategy
+    controller = deployed.controller
+    settKeeper = accounts.at(sett.keeper(), force=True)
+
+    snap = SnapshotManager(sett, strategy, controller, "StrategySnapshot")
+    randomUser = accounts[6]
+    # End Setup
+
+    # Deposit
+    assert want.balanceOf(deployer) > 0
+
+    depositAmount = int(want.balanceOf(deployer) * 0.8)
+    assert depositAmount > 0
+
+    want.approve(sett.address, MaxUint256, {"from": deployer})
+
+    snap.settDeposit(depositAmount, {"from": deployer})
+
+    shares = sett.balanceOf(deployer)
+
+    # Earn
+    with brownie.reverts("onlyAuthorizedActors"):
+        sett.earn({"from": randomUser})
+
+    snap.settEarn({"from": settKeeper})
+
+    chain.sleep(5000)
+    chain.mine(1)
+
+    ## Before Harvest, check the initial bveCVX balance
+    tree = strategy.BADGER_TREE()
+    veCVX = interface.ISett(strategy.CVX_VAULT())
+    beforeBalance = veCVX.balanceOf(tree)
+    
+    ## Harvest
+    snap.settHarvest({"from": settKeeper})
+
+    ## Check that Badger Tree has more bveCVX
+    assert veCVX.balanceOf(tree) > beforeBalance
+    assert history[-1].events["TreeDistribution"]["amount"] == veCVX.balanceOf(tree) - beforeBalance
```

Comment:
This test lgtm but imo it would be better for the repo's resilience to handle these checks at the resolver level. I would suggest that we add the badgerTree to the entities and bveCVX to the tracked tokens on it. Having the two assertions above on the resolver would remove the need of this new suite as they would be covered by the `tests/test_harvest_flow.py`.

Reply:
Moved testing to resolver
<img width="894" alt="Screenshot 2022-01-06 at 14 39 42" src="https://user-images.githubusercontent.com/13383782/148391487-fd23314e-a523-42ec-9d71-eac2a4de2bc8.png">